rest/rest-resources/src/test/java/org/apache/brooklyn/rest/resources/CatalogResourceTest.java

```diff
@@ -487,6 +495,79 @@ public void testAddMissingItem() {
         //equivalent to HTTP response 404 text/html
         addAddCatalogItemWithInvalidBundleUrl("classpath://missing-jar-file.txt");
     }
+    
+    @Test
+    public void testOsgiBundleWithBom() throws Exception {
+        TestResourceUnavailableException.throwIfResourceUnavailable(getClass(), OsgiStandaloneTest.BROOKLYN_TEST_OSGI_ENTITIES_PATH);
```

Comment:
The `CatalogBomScanner` automatically adds library information to the `catalog.bom` about the bundle that contains it; but see my other comments about `CatalogBomScanner` and deleting the temp files.